src/sources/http.rs

```diff
@@ -73,7 +85,7 @@ impl SourceConfig for SimpleHttpConfig {
             encoding: self.encoding,
             headers: self.headers.clone(),
         };
-        source.run(self.address, "", &self.tls, out, shutdown)
+        source.run(self.address, "", &self.tls, out, shutdown).await
```

Comment:
This doesn't quite seem right. We're going to `await` the return value of `build_async` when we're building the topology, and this seems to imply that it will immediately start running and not return until it's done processing the input.

Reply:
This partially incorrect, I mentioned this in https://github.com/timberio/vector/pull/3188#discussion_r461653474.
`run` is async because bind in tokio:0.2 is async, previously bind happened on topology building as sync. `run` result is still future: https://github.com/timberio/vector/blob/9e13da7960a2ecca0878d38fd15ddf65ad23dae7/src/sources/util/http.rs#L120-L130
I remember that `future01::lazy` was used in other sinks/sources for moving bind out of topology building, but here was no lazy usage, so everything saved. Should I move bind out of topology building?

Reply:
Mentioned this in slack, but for the record we should not bind within `build`, but only when the future returned by `build` is polled.